### PR TITLE
MULE-17058: CXF Proxy service losing envelope data with user-defined interceptors

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

File: modules/cxf/src/test/resources/log4j2-test.xml [hunk 1]
```diff
@@ -20,6 +20,7 @@
         <!-- Mule classes -->
         <AsyncLogger name="org.mule" level="WARN"/>
         <AsyncLogger name="com.mulesoft" level="WARN"/>
```
Comment (fsgonz) [inline modules/cxf/src/test/resources/log4j2-test.xml]: Is this needed in the test?

File: modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java [hunk 2]
```diff
@@ -156,4 +160,12 @@ protected String getMimeType()
         return MimeTypes.XML;
     }
 
+    @Override
+    protected void addInInterceptorCorrectingInterceptors(ServerFactoryBean sfb)
+    {
+        if (isProxy() && isProxyEnvelope() && CxfUtils.interceptorOfClassIsPresent(sfb.getInInterceptors(), SAAJInInterceptor.class))
```
Comment (fsgonz) [inline modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java]: static import

Comment (fsgonz) [inline modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java]: Is this only needed for the case of SAAJInInterceptor.

File: modules/cxf/src/test/resources/cxf-proxy-service-tests-config.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <cxf:proxy-service port="HelloServicePort"
+                           namespace="http://test.Pablo.name/"
+                           service="HelloServiceService"
+                           payload="envelope" enableMuleSoapHeaders="false"
+                           wsdlLocation="classpath:${wsdl.uri}">
+            <cxf:inInterceptors>
+                <spring:bean class="org.apache.cxf.binding.soap.saaj.SAAJInInterceptor"/>
+            </cxf:inInterceptors>
+        </cxf:proxy-service>
+        <logger message="#['Payload AFTER proxy-service: ' + message.payloadAs(java.lang.String)]"
```
Comment (fsgonz) [inline modules/cxf/src/test/resources/cxf-proxy-service-tests-config.xml]: remove loggers in test.

File: modules/cxf/src/test/resources/log4j2-test.xml (hunk 1 above)
Comment (fsgonz) [reply modules/cxf/src/test/resources/log4j2-test.xml]: Remove loggers in test.

File: modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java (hunk 2 above)
Comment (plbalbi) [reply modules/cxf/src/main/java/org/mule/module/cxf/builder/ProxyServiceMessageProcessorBuilder.java]: As far I looked into it, yes. Either way, the interceptor check if in the CXF's message contents there is a SOAPMessage, from which it regenerates the xml stream with the envelope

**Pull request merged**